template/app/src/main/java/co/nimblehq/template/ui/ErrorMapping.kt

```diff
@@ -2,7 +2,11 @@ package co.nimblehq.template.ui
 
 import android.content.Context
 import co.nimblehq.template.R
+import co.nimblehq.template.data.service.ApiException
 
 fun Throwable.userReadableMessage(context: Context): String {
-    return context.getString(R.string.error_generic)
+    return when (this) {
+        is ApiException -> error?.message
+        else -> message
```

Comment:
[Optional] Shall we also add a case for `NoConnectivityException`? 😄 
```suggestion
    return when (this) {
        is ApiException -> error?.message
        is NoConnectivityException -> context.getString(R.string.error_no_connectivity)
        else -> message
```

Reply:
@ryan-conway I would like to keep the pure concept of the template. 😄